core/storage/btree.rs

```diff
@@ -1431,7 +1558,7 @@ impl BTreeCursor {
                         }
                     }
                     let rightmost_page_id = *rightmost_page_id;
-                    let c = self.move_to_root()?;
+                    let c = return_if_io!(self.move_to_root());
```

Comment:
How does this work? If we yield on IO here, then move_to_right_state will not be changed. If we don't yield on IO here, then is `c` ever `Some`?